tutorial/10_key_features/003_efficient_optimization_algorithms.py

```diff
@@ -0,0 +1,146 @@
+study = optuna.create_study(pruner=optuna.pruners.MedianPruner())
+study.optimize(objective, n_trials=20)
+
+###################################################################################################
+# As you can see, several trials were pruned (stopped) before they finished all of the iterations.
+# The format of message is ``"Trial <Trial Number> pruned."``.
+
+###################################################################################################
+# Which Sampler and Pruner Should be Used?
+# ----------------------------------------
+#
+# From the benchmark results which are available at `optuna/optuna - wiki "Benchmarks with Kurobako" <https://github.com/optuna/optuna/wiki/%5BUnder-Construction%5D-Benchmarks>`_, at least for not deep learning tasks, we would say that
+#
+# * For :class:`optuna.samplers.RandomSampler`, :class:`optuna.pruners.MedianPruner` is the best.
+# * For :class:`optuna.samplers.TPESampler`, :class:`optuna.pruners.Hyperband` is the best.
+#
+# However, note that the benchmark is not deep learning.
+# For deep learning tasks,
+# consult the below table from `Ozaki et al, Hyperparameter Optimization Methods: Overview and Characteristics, in IEICE Trans, Vol.J103-D No.9 pp.615-631, 2020 <https://doi.org/10.14923/transinfj.2019JDR0003>`_,
+#
+# +---------------------------+-----------------------------------------+---------------------------------------------------------------+
+# | Parallel Compute Resource | Categorical/Conditional Hyperparameters | Recommended Algorithms                                        |
+# +===========================+=========================================+===============================================================+
+# | Limited                   | No                                      | TPE. GP-EI if search space is low-dimensional and continuous. |
+# +                           +-----------------------------------------+---------------------------------------------------------------+
+# |                           | Yes                                     | TPE. GP-EI if search space is low-dimensional and continuous  |
+# +---------------------------+-----------------------------------------+---------------------------------------------------------------+
+# | Sufficient                | No                                      | CMA-ES, Random Search                                         |
+# +                           +-----------------------------------------+---------------------------------------------------------------+
+# |                           | Yes                                     | Random Search or Genetic Algorithm                            |
+# +---------------------------+-----------------------------------------+---------------------------------------------------------------+
+#
+
+###################################################################################################
+# Integration Modules for Pruning
```

Comment:
The integration modules are indeed relevant to the pruning but I'm wondering if it doesn't stretch the scope of this section. It might be a bit overwhelming for targeted readers. Actually, can we not provide a separate recipe for Optuna's various integration features?

Reply:
In my understanding, the way of pruning trials in this file rather confuses readers about how to use this feature with kind of high-level libraries like Keras, Lightning, and Ignite. So just saying we have integration modules is worth.

Reply:
Should clarify integration modules are convenient for the libraries such as Keras and PyTorch Lightning that abstract the training loop.

Reply:
Alright, let's leave it as is. We could however place this section before "Which Sampler and Pruner Should be Used?", which also addresses samplers.